recipes/freeglut/all/conanfile.py

```diff
@@ -0,0 +1,124 @@
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = "FreeGLUT-FG_" + self.version.replace(".", "_")
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        # See https://github.com/dcnieho/FreeGLUT/blob/44cf4b5b85cf6037349c1c8740b2531d7278207d/README.cmake
+        cmake = CMake(self, set_cmake_flags=True)
+
+        cmake.definitions["FREEGLUT_BUILD_DEMOS"] = False
+        cmake.definitions["FREEGLUT_BUILD_STATIC_LIBS"] = not self.options.shared
+        cmake.definitions["FREEGLUT_BUILD_SHARED_LIBS"] = self.options.shared
+        cmake.definitions["FREEGLUT_GLES"] = self.options.gles
+        cmake.definitions["FREEGLUT_PRINT_ERRORS"] = self.options.print_errors_at_runtime
+        cmake.definitions["FREEGLUT_PRINT_WARNINGS"] = self.options.print_warnings_at_runtime
+        cmake.definitions["FREEGLUT_INSTALL_PDB"] = False
+        cmake.definitions["INSTALL_PDB"] = False
+        # cmake.definitions["FREEGLUT_WAYLAND"] = "ON" if self.options.wayland else "OFF" # nightly version only as of now
+
+        cmake.configure(build_folder=self._build_subfolder)
+        return cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        self.copy(pattern="COPYING", dst="licenses", src=self._source_subfolder)
+        cmake = self._configure_cmake()
+        cmake.install()
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
+
+    def package_info(self):
+        self.cpp_info.filenames["cmake_find_package"] = "FreeGLUT"
+        self.cpp_info.filenames["cmake_find_package_multi"] = "FreeGLUT"
+        self.cpp_info.names["cmake_find_package"] = "FreeGLUT"
+        self.cpp_info.names["cmake_find_package_multi"] = "FreeGLUT"
+
+        self.cpp_info.components['freeglut_'].libs = tools.collect_libs(self)
+
+        self.cpp_info.components['freeglut_'].names["cmake_find_package"] = "freeglut" if self.options.shared else "freeglut_static"
+        self.cpp_info.components['freeglut_'].names["cmake_find_package_multi"] = "freeglut" if self.options.shared else "freeglut_static"
+        self.cpp_info.components['freeglut_'].names["pkg_config"] = "freeglut" if self.settings.os == "Windows" else "glut"
+
+        self.cpp_info.components['freeglut_'].requires.append("opengl::opengl")
+        self.cpp_info.components['freeglut_'].requires.append('glu::glu')
+        if self.settings.os == "Linux":
+            self.cpp_info.components['freeglut_'].requires.append("xorg::xorg")
+
+        if self.settings.os == "Windows":
+            if not self.options.shared:
+                self.cpp_info.components['freeglut_'].defines.append("FREEGLUT_STATIC=1")
+            self.cpp_info.components['freeglut_'].defines.append("FREEGLUT_LIB_PRAGMAS=0")
```

Comment:
Is this macro also required for mingw?
`#pragma warning` is only used by msvc.

Reply:
might be required for Intel/Clang, not sure about MinGW